linetools/isgm/abssystem.py

```diff
@@ -0,0 +1,202 @@
+from astropy.table import QTable
+from astropy import constants as const
+from astropy.coordinates import SkyCoord
+
+from linetools.isgm.abscomponent import AbsComponent
+
+class AbsSystem(object):
+    """
+    Class for an absorption line system
+
+    Parameters
+    ----------
+    abs_type : str
+      Type of absorption system, e.g. MgII, DLA, LLS
+    radec : tuple or coordinate
+        RA/Dec of the sightline or astropy.coordinate
+    zabs : float
+      Absorption redshift
+    vlim : Quantity array (2)
+      Velocity limits of the system
+    zem : float, optional
+      Emission redshift of background source
+    NHI :  float, optional
+      Log10 of the HI column density
+    sigNHI :  np.array(2), optional
+      Log10 error of the HI column density (-/+)
+    name : str, optional
+      Name for the system
+
+    Attributes
+    ----------
+        abs_type : str
+        coord : SkyCoord
+            RA/Dec of the sightline
+        zabs : float
+          Absorption redshift
+        zem : float
+          Emission redshift of background source
+        vlim : Quantity array (2)
+          Velocity limits of the system
+        NHI :  float
+          Log10 of the HI column density
+        sigNHI :  np.array(2)
+          Log10 error of the HI column density (-/+)
+        MH :  float
+          Metallicity (log10)
+        name : str
+            Name of the system
+    """
+
+    __metaclass__ = ABCMeta
+
+    @classmethod
+    def from_components(cls, components):
+        """Instantiate from a list of AbsComponent objects
+
+        Parameters
+        ----------
+        components : list
+          List of AbsComponent objects
+        """
+        # Check
+        if not isinstance(components,list):
+            raise IOError('Need a list of AbsComponent objects')
+        if not all(isinstance(x,AbsComponent) for x in components):
+            raise IOError('List needs to contain AbsLine objects')
+        # Instantiate with the first component
+        init_comp = components[0]
+        slf = cls(init_comp.coord, init_comp.zcomp, init_comp.vlim)
+        if slf.chk_component(init_comp):
+            slf._components.append(init_comp)
+        else:
+            raise IOError("Bad component input")
+        # Append with component checking
+        if len(components) > 1:
+            for component in components[1:]:
+                slf.add_component(component)
+        # Return
+        return slf
+
+    def __init__(self, abs_type, radec, zabs, vlim, zem=0., NHI=0., sigNHI=np.zeros(2), name=''):
+
+        self.zabs = zabs
+        self.zem = zem
+        self.vlim = vlim
+        self.NHI = NHI
+        self.sigNHI = sigNHI
+        # RA/DEC
+        if isinstance(radec,(tuple)):
+            self.coord = SkyCoord(ra=radec[0], dec=radec[1])
+        elif isinstance(radec,SkyCoord):
+            self.coord = radec
+        self.name = name
+
+        # Abs type
+        if abs_type == None:
```

Comment:
This seems it may break, because abs_type is not defined. Again, should we have this as an optional attribute?